eZ/Publish/Core/Persistence/Legacy/Filter/CriterionQueryBuilder/Content/ObjectStateIdQueryBuilder.php

```diff
@@ -29,19 +29,21 @@ public function buildQueryConstraint(
         FilteringQueryBuilder $queryBuilder,
         FilteringCriterion $criterion
     ): ?string {
+        $tableAlias = uniqid('osl_');
+
         /** @var \eZ\Publish\API\Repository\Values\Content\Query\Criterion\ObjectStateId $criterion */
         $queryBuilder
             ->joinOnce(
```

Comment:
Consider it as a non merge blocking comment:

Just a remark - if we will always generate custom table alias for each `ObjectStateId` criteria, then it would make sense to switch into simpler `QueryBuilder::join`, which does not contain additional checks against table uniqueness.